app/models/concerns/orchestration/dns.rb

```diff
@@ -2,179 +2,224 @@ module Orchestration::DNS
   extend ActiveSupport::Concern
   include Orchestration::Common
 
+  RECORD_TYPES = [:a, :aaaa, :ptr, :ptr6]
+
   included do
     after_validation :dns_conflict_detected?, :queue_dns, :unless => :importing_facts
     before_destroy :queue_dns_destroy, :unless => :importing_facts
     register_rebuild(:rebuild_dns, N_('DNS'))
   end
 
-  def dns?
+  def dns_ready?
     # host.managed? and managed? should always come first so that orchestration doesn't
     # even get tested for such objects
-    (host.nil? || host.managed?) && managed? && hostname.present? && ip_available? && !domain.nil? && !domain.proxy.nil? && SETTINGS[:unattended]
+    SETTINGS[:unattended] && (host.nil? || host.managed?) && managed? && hostname.present?
+  end
+
+  def dns?
+    dns_ready? && ip_available? && !domain.nil? && !domain.proxy.nil?
```

Comment:
nitpick, `variable.present?` seems more readable than `!variable.nil?` - applies to many lines around this one